### ISSUE TITLE
AC Cartpole: I think the better loss function is this one.

### ISSUE THREAD

**Issue by zsdonghao** [MorvanZhou/Reinforcement-learning-with-tensorflow#6]
```python
cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=self.acts_logits, logits=self.a, name='ce')
loss = tf.reduce_sum(tf.multiply(cross_entropy, self.td_error))
```

**Comment by MorvanZhou** [comment]
@zsdonghao I think they are both equivalent. In here https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/master/contents/7_Policy_gradient_softmax/RL_brain.py. 
`with tf.name_scope('loss'):
            # to maximize total reward (log_p * R) is to minimize -(log_p * R), and the tf only have minimize(loss)
            neg_log_prob = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=all_act, labels=self.tf_acts)   # this is negative log of chosen action
            # or in this way:
            # neg_log_prob = tf.reduce_sum(-tf.log(self.all_act_prob)*tf.one_hot(self.tf_acts, self.n_actions), axis=1)
            loss = tf.reduce_mean(neg_log_prob * self.tf_vt)  # reward guided loss`
I summarize this a little bit.
